src/cycles/common.ts

```diff
@@ -0,0 +1,22 @@
+import { NodeInfo } from '../core/types';
```

Comment:
Can we call this file something else? Anything else 🙏  Maybe `graph.ts`? Is it even actually needed?